hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -231,7 +231,13 @@ private static Option<Schema.Field> findNestedField(Schema schema, String[] fiel
     if (!nestedPart.isPresent()) {
       return Option.empty();
     }
-    return nestedPart;
+    boolean isUnion = false;
```

Comment:
Could you write a unit test around the new logic (make the method accessible by the test class)?

Reply:
hudi-common/src/test/java/org/apache/hudi/avro/TestAvroSchemaUtils.java I uncommented the test in this pr